packages/design-system/src/components/OcResourceIcon/OcResourceIcon.vue

```diff
@@ -122,4 +129,83 @@ export default defineComponent({
     height: 70%;
   }
 }
+.oc-resource-icon-status-badge {
+  position: absolute;
+  bottom: -3px;
+  right: -2px;
+  width: 12px;
+  height: 12px;
+  padding: var(--oc-space-xsmall);
+  border-radius: 30px;
+  transition: background-color 200ms ease-in-out;
+  background: var(--oc-color-background-default);
+
+  &-inner {
+    position: absolute;
+    top: 2px;
+    left: 2px;
+    width: 8px;
+    height: 8px;
+    padding: var(--oc-space-xsmall);
+    line-height: var(--oc-space-small);
+    border-radius: 30px;
+  }
+  .oc-icon {
+    margin-top: -2px;
+    margin-left: -1.5px;
+    svg {
+      fill: var(--oc-color-background-default) !important;
+    }
+  }
+  .oc-spinner {
+    margin-left: -2px;
+    margin-top: -2px;
+    color: var(--oc-color-background-default) !important;
+  }
+}
+// different background for tiles view
+.oc-tile-card-preview {
```

Comment:
not sure if this is the right place for the styling but for me it makes sense to put it where the badge is actually implemented vs. adding the style to the tiles view

Reply:
I get your point. But technically the view doesn't know anything about the tiles. I would rather add the code there